### ISSUE TITLE
Adding basic calculus functions

### ISSUE THREAD

**Issue by supreet11agrawal** [sympy/sympy#16520]
See https://github.com/sympy/sympy/pull/16473
A few functions were introduced in this PR. The work here will be to include more functions which are quite basic yet useful in calculus.
The work can be listed as:
* [ ] Implement a function, maybe `stationary_values` which returns a list of stationary points and corresponding values.
* [ ] Implement a function, `critical_points`( and maybe `critical_values` as well) which returns set of critical points.
* [ ] Implement a function, `minima` which returns minima of an expression in the given domain.
* [ ] Implement a function, `maxima` which returns maxima of an expression in the given domain.
* [ ] Implement a function, `inflection_points` which returns the inflection points of an expression in the given domain.
* [ ] Similarly adding `inflection_values` for list of points and corresponding function values.

**Comment by janvi7109** [comment]
I would like to work on this issue

**Comment by supreet11agrawal** [comment]
Yes, sure! Go ahead

**Comment by epistemologist** [comment]
Is it okay if I also work on this issue?

**Comment by SongyuanGuan** [comment]
Hi, I am very interested in this project. May I also work on this issue？

**Comment by oscarbenjamin** [comment]
I think that there should be agreement about the signature/API of these functions here in this issue before PRs are opened.

For example argmax and argmin can be used more generally than just for differentiable functions of a single real argument. Other concepts like stationary points should in principle be definable for functions of multiple variables so I would prefer not to lock in an API that doesn't work for that case. (This also applies to the functions already added in #16473).

These functions are likely to end up being in the main sympy namespace which makes it hard to change them for backwards compatibility. When introducing new functions like this we should treat it as if we will only *ever* get one chance to define a function of that name: if we will want it to do more general things in future then the signature should be chosen appropriately *now* rather than later.

**Comment by supreet11agrawal** [comment]
@oscarbenjamin I totally agree with you on this point. Perhaps we should discuss what should be done about these functions. I think they are pretty useful for a CAS and thus should be implemented ASAP. I remember that a few questions on **Stackoverflow** were specifically related to these functions.
Eg.
* [How to find the domain of a function in python using sympy](https://stackoverflow.com/questions/55834798/how-to-find-the-domain-of-a-function-in-python-using-sympy/55885326#55885326) 

* [How to know whether a function is continuous with sympy?](https://stackoverflow.com/questions/55881078/how-to-know-whether-a-function-is-continuous-with-sympy/55885282#55885282) 

**CC** @asmeurer @smichr

**Comment by oscarbenjamin** [comment]
@supreet11agrawal I agree that these are important functions: that's why it is even more important that we get them right.

Why don't you survey what other CASs do and propose an API here?

For example (and related functions in see also):
https://reference.wolfram.com/language/ref/Maximize.html
